### ISSUE TITLE
i can't build it

### ISSUE THREAD

**Issue by jeckdz** [gabime/spdlog#2220]
it show "\core.h(2729): error C2131: 表达式的计算结果不是常数" when i build it by vs 2017.

**Comment by tt4g** [comment]
It is difficult to help you because there is too little information provided.
Please provide build commands and complete compilation error messages.

**Comment by jeckdz** [comment]
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): error C2131: 表达式的计算结果不是常数
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): note: 遇到非常量(子)表达式
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2848): note: 参见对正在编译的函数 模板 实例化“void fmt::v8::detail::check_format_string<fmt::v8::string_view&,const char(&)[3],S,0>(S)”的引用
1>        with
1>        [
1>            S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING
1>        ]
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用
1>        with
1>        [
1>            S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING
1>        ]
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用
1>        with
1>        [
1>            S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING
1>        ]
1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): fatal error C1903: 无法从以前的错误中恢复；正在停止编译

**Comment by jeckdz** [comment]
1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2065: “Lhost”: 未声明的标识符
1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2228: “.c_str”的左边必须有类/结构/联合
1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): note: 类型是“unknown-type”
1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2660: “InetPtonW”: 函数不接受 2 个参数
![image](https://user-images.githubusercontent.com/42936747/147220084-cd5319e2-778b-4b94-b969-602fa48403b1.png)

**Comment by jeckdz** [comment]
i build it by Microsoft Visual Studio 2017. There are two issures when i copy the include/spdlog to my project.

**Comment by tt4g** [comment]
To https://github.com/gabime/spdlog/issues/2220#issuecomment-1000166996

> 1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2065: “Lhost”: 未声明的标识符 1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2228: “.c_str”的左边必须有类/结构/联合 1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): note: 类型是“unknown-type” 1>C:\Users\235957\Desktop\test\test\test\spdlog/details/udp_client-windows.h(73): error C2660: “InetPtonW”: 函数不接受 2 个参数 ![image](https://user-images.githubusercontent.com/42936747/147220084-cd5319e2-778b-4b94-b969-602fa48403b1.png)

This error has nothing to do with spdlog.

You are using the `TEXT` macro incorrectly.
The `TEXT` macro is used to map a string literal to either `char *` or `wchar_t *` according to the project configuration.
It cannot be used for `std::string::c_str()`.

**Comment by tt4g** [comment]
To: https://github.com/gabime/spdlog/issues/2220#issuecomment-1000162902

> 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): error C2131: 表达式的计算结果不是常数 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): note: 遇到非常量(子)表达式 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2848): note: 参见对正在编译的函数 模板 实例化“void fmt::v8::detail::check_format_string<fmt::v8::string_view&,const char(&)[3],S,0>(S)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): fatal error C1903: 无法从以前的错误中恢复；正在停止编译

This error reports a bug in the source code where the number of log arguments and placeholders do not match.

```cpp
spdlog::info("{}"); // Error
spdlog::info("1: {} 2: {}", "foo"); // Error
spdlog::info("{:d}", "not a number"); // Error

spdlog::info("{}", "foo"); // OK
spdlog::info("1: {} 2: {}", "foo", "bar"); // OK
spdlog::info("{:d}", 10); // OK
```

Check all the places where you are logging with spdlog and carefully check that the number of log arguments and placeholders match.

**Comment by jeckdz** [comment]
> To: [#2220 (comment)](https://github.com/gabime/spdlog/issues/2220#issuecomment-1000162902)
> 
> > 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): error C2131: 表达式的计算结果不是常数 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): note: 遇到非常量(子)表达式 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2848): note: 参见对正在编译的函数 模板 实例化“void fmt::v8::detail::check_format_string<fmt::v8::string_view&,const char(&)[3],S,0>(S)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\format-inl.h(74): note: 参见对正在编译的函数 模板 实例化“fmt::v8::basic_format_string<char,fmt::v8::string_view &,const char (&)[3]>::basic_format_string<fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING,0>(const S &)”的引用 1> with 1> [ 1> S=fmt::v8::detail::format_error_code::<lambda_6f3312de3d5ecb9dcf1123b9a3829092>::()::FMT_COMPILE_STRING 1> ] 1>c:\users\235957\desktop\test\test\test\spdlog\fmt\bundled\core.h(2729): fatal error C1903: 无法从以前的错误中恢复；正在停止编译
> 
> This error reports a bug in the source code where the number of log arguments and placeholders do not match.
> 
> ```c++
> spdlog::info("{}"); // Error
> spdlog::info("1: {} 2: {}", "foo"); // Error
> spdlog::info("{:d}", "not a number"); // Error
> 
> spdlog::info("{}", "foo"); // OK
> spdlog::info("1: {} 2: {}", "foo", "bar"); // OK
> spdlog::info("{:d}", 10); // OK
> ```
> 
> Check all the places where you are logging with spdlog and carefully check that the number of log arguments and placeholders match.

I only #include "spdlog/spdlog.h"  and do nothing

**Comment by tt4g** [comment]
> I only #include "spdlog/spdlog.h" and do nothing

This may be a bug that was discussed in #2081.

This bug is already fixed in the current branch (fmtlib/fmt#2482), but it is not yet released.
The same fix as this commit fmtlib/fmt@2520f410c82e7e7b630f39a6d9164e4bcbc029cd may work.

**Comment by jeckdz** [comment]
> > I only #include "spdlog/spdlog.h" and do nothing
> 
> This may be a bug that was discussed in #2081.
> 
> This bug is already fixed in the current branch ([fmtlib/fmt#2482](https://github.com/fmtlib/fmt/pull/2482)), but it is not yet released. The same fix as this commit [fmtlib/fmt@2520f41](https://github.com/fmtlib/fmt/commit/2520f410c82e7e7b630f39a6d9164e4bcbc029cd) may work.

OK,I will check it. thank you very much.
